contributors/design-proposals/pod-priority-api.md

```diff
@@ -0,0 +1,228 @@
+what determines the order at which pods are scheduled are beyond the scope of
+this document.
+
+### Effect of priority on preemption 
+
+Generally, lower priority pods are more likely to get preempted by higher
+priority pods when cluster has reached a threshold. In such a case, scheduler
+may decide to preempt lower priority pods to release enough resources for higher
+priority pending pods. As mentioned before, there are many other parameters
+that affect scheduling decisions, such as affinity and anti-affinity. If
+scheduler determines that a high priority pod cannot be scheduled even if lower
+priority pods are preempted, it will not preempt lower priority pods. Scheduler
+may have other restrictions on preempting pods, for example, it may refuse to
+preempt a pod if PodDisruptionBudget is violated. The details of scheduling and
+preemption decisions are beyond the scope of this document.
+
+### Priority in PodSpec 
+
+Pods may have priority in their pod spec. PodSpec will have two new fields
+called "PriorityClassName" which is specified by user, and "Priority" which will
+be populated by Kubernetes. User-specified priority (PriorityClassName) is a 
+string and all of the valid priority classes are defined by a system wide
+mapping that maps each string to an integer. The PriorityClassName specified in
+a pod spec must be found in this map or the pod creation request will be
+rejected (See below for more info on name resolution). If PriorityClassName is
+empty, it will resolve to priority zero. Once the PriorityClassName is resolved
+to an integer, it is placed in "Priority" field of PodSpec.
+
+
+```
+type PodSpec struct {
+  ...
+  PriorityClassName string
+  Priority          int32  // Populated by Admission Controller. Users are not allowed to set it directly.
+}
+```
+
+### Priority Classes 
+
+A Kubernetes cluster has one predefined (built-in) priority class which is 
+reserved for the system pods and its value is the largest 32-bit
+integer. The cluster may have many more user defined priority classes for
+various use cases. The following list is an example of how the priorities and
+their values may look like.
+We may decide to not ship Kubernetes with predefined priority classes, except
+for the "system" priority which will be built in Kubernetes. The main goal of
+of having no built-in priority classes is to avoid creating defacto standard
+names which may be hard to change in the future. The big disadvantage of not
+having built-in priority classes is losing config portability. PodSpecs written
+for one cluster would most like not work on a different cluster.
+(We will discuss this further and finalize the decision.)
+
+
+```
+system  2147483647 (int_max)
+tier1   4000
+tier2   2000
+tier3   1000
+```
+
+The following shows a list of example workloads in a Kubernetes cluster in decreasing order of priority:
+
+* Kubernetes system daemons (per-node like fluentd, and cluster-level like 
```

Comment:
This is from the link that you sent:

> A job priority can be any integer, and larger values are of higher priority. So, 0 is a higher job priority than -3, and 6 is a higher job priority than 5.

Based on this design proposal, defining priority classes is a one-time operation and after that users (and admins) will work mostly with the priority class names and not with the actual integer value of the priority. So, hopefully they shouldn't worry much about the actual integer value.
Besides, if we wanted to use lower values as higher priority, we would have had to choose a different name for the field instead of `priority`, maybe something like `nice`. When "niceness" is lower, it makes sense for the process to not yield to other processes, but it does not make sense when a lower value for a field called `priority` means higher priority.